core/node/node_framework/src/implementations/resources/eth_interface.rs

```diff
@@ -13,9 +13,23 @@ impl Resource for EthInterfaceResource {
     }
 }
 
+/// A resource that provides L2 interface object to the service.
+/// It is expected to have the same URL as the `EthInterfaceResource`, but have different capabilities.
+///
+/// This resource is provided separately from `EthInterfaceResource`, to provide type safety in places, where the
+/// component must work with L1-interface only and should use `EthInterfaceResource` instead.
+#[derive(Debug, Clone)]
+pub struct L2InterfaceResource(pub Box<DynClient<L2>>);
```

Comment:
Bikeshedding: Maybe, call this `L2SettlementInterfaceResource`, so that the intended use case of this resource is clear? (Or simply `L2SettlementResource`; I find the "interface" part in `EthInterface` / `BoundEthInterface` somewhat non-descriptive.)

Reply:
I like better what is here now, so naming based on "what it is" rather "what it's used for", I also really like the "Interface" in the names, it tells me instantly that it's some sort of API resource

Reply:
Naming is subjective; IMO, `_Client` better signals being an API resource 🤷 Also, why is this convention not followed for `EthInterfaceResource` then? If `GasAdjusterClient` is refactored as per another comment, it could be reworked to wrap `EthFeeInterface`, and an additional L2 client resource wouldn't be necessary IIUC.